diff --git a/node_modules/expo-local-authentication/android/.gradle/8.3/checksums/checksums.lock b/node_modules/expo-local-authentication/android/.gradle/8.3/checksums/checksums.lock
new file mode 100644
index 0000000..014ad8b
Binary files /dev/null and b/node_modules/expo-local-authentication/android/.gradle/8.3/checksums/checksums.lock differ
diff --git a/node_modules/expo-local-authentication/android/.gradle/8.3/dependencies-accessors/dependencies-accessors.lock b/node_modules/expo-local-authentication/android/.gradle/8.3/dependencies-accessors/dependencies-accessors.lock
new file mode 100644
index 0000000..5aefa82
Binary files /dev/null and b/node_modules/expo-local-authentication/android/.gradle/8.3/dependencies-accessors/dependencies-accessors.lock differ
diff --git a/node_modules/expo-local-authentication/android/.gradle/8.3/dependencies-accessors/gc.properties b/node_modules/expo-local-authentication/android/.gradle/8.3/dependencies-accessors/gc.properties
new file mode 100644
index 0000000..e69de29
diff --git a/node_modules/expo-local-authentication/android/.gradle/8.3/fileChanges/last-build.bin b/node_modules/expo-local-authentication/android/.gradle/8.3/fileChanges/last-build.bin
new file mode 100644
index 0000000..f76dd23
Binary files /dev/null and b/node_modules/expo-local-authentication/android/.gradle/8.3/fileChanges/last-build.bin differ
diff --git a/node_modules/expo-local-authentication/android/.gradle/8.3/fileHashes/fileHashes.lock b/node_modules/expo-local-authentication/android/.gradle/8.3/fileHashes/fileHashes.lock
new file mode 100644
index 0000000..e97760a
Binary files /dev/null and b/node_modules/expo-local-authentication/android/.gradle/8.3/fileHashes/fileHashes.lock differ
diff --git a/node_modules/expo-local-authentication/android/.gradle/8.3/gc.properties b/node_modules/expo-local-authentication/android/.gradle/8.3/gc.properties
new file mode 100644
index 0000000..e69de29
diff --git a/node_modules/expo-local-authentication/android/.gradle/vcs-1/gc.properties b/node_modules/expo-local-authentication/android/.gradle/vcs-1/gc.properties
new file mode 100644
index 0000000..e69de29
diff --git a/node_modules/expo-local-authentication/android/src/main/java/expo/modules/localauthentication/LocalAuthenticationModule.kt b/node_modules/expo-local-authentication/android/src/main/java/expo/modules/localauthentication/LocalAuthenticationModule.kt
index 6390b58..082edad 100644
--- a/node_modules/expo-local-authentication/android/src/main/java/expo/modules/localauthentication/LocalAuthenticationModule.kt
+++ b/node_modules/expo-local-authentication/android/src/main/java/expo/modules/localauthentication/LocalAuthenticationModule.kt
@@ -240,7 +240,7 @@ class LocalAuthenticationModule : Module() {
         setNegativeButtonText(cancelLabel)
       } else {
         setAllowedAuthenticators(
-          BiometricManager.Authenticators.BIOMETRIC_WEAK
+          BiometricManager.Authenticators.BIOMETRIC_STRONG
             or BiometricManager.Authenticators.DEVICE_CREDENTIAL
         )
       }
@@ -350,7 +350,7 @@ class LocalAuthenticationModule : Module() {
   }
 
   private fun canAuthenticateUsingWeakBiometrics(): Int =
-    biometricManager.canAuthenticate(BiometricManager.Authenticators.BIOMETRIC_WEAK)
+    biometricManager.canAuthenticate(BiometricManager.Authenticators.BIOMETRIC_STRONG)
 
   private fun createResponse(
     error: String? = null,
