diff --git a/node_modules/react-native-tcp-socket/ios/TcpSockets.m b/node_modules/react-native-tcp-socket/ios/TcpSockets.m
index fc99b73..6d57ce1 100644
--- a/node_modules/react-native-tcp-socket/ios/TcpSockets.m
+++ b/node_modules/react-native-tcp-socket/ios/TcpSockets.m
@@ -219,20 +219,22 @@ RCT_EXPORT_METHOD(getCertificate:(nonnull NSNumber *)cId
 
 - (void)onConnect:(TcpSocketClient *)client {
     GCDAsyncSocket *socket = [client getSocket];
-    [self sendEventWithName:@"connect"
-                       body:@{
-                           @"id" : client.id,
-                           @"connection" : @{
-                               @"localAddress" : [socket localHost],
-                               @"localPort" :
-                                   [NSNumber numberWithInt:[socket localPort]],
-                               @"remoteAddress" : [socket connectedHost],
-                               @"remotePort" : [NSNumber
-                                   numberWithInt:[socket connectedPort]],
-                               @"remoteFamily" : [socket isIPv4] ? @"IPv4"
-                                                                 : @"IPv6"
-                           }
-                       }];
+    if([socket localPort] != nil && [socket connectedPort] != nil) {
+      [self sendEventWithName:@"connect"
+                         body:@{
+                             @"id" : client.id,
+                             @"connection" : @{
+                                 @"localAddress" : [socket localHost],
+                                 @"localPort" :
+                                     [NSNumber numberWithInt:[socket localPort]],
+                                 @"remoteAddress" : [socket connectedHost],
+                                 @"remotePort" : [NSNumber
+                                     numberWithInt:[socket connectedPort]],
+                                 @"remoteFamily" : [socket isIPv4] ? @"IPv4"
+                                                                   : @"IPv6"
+                             }
+                         }];
+    }
 }
 
 - (void)onListen:(TcpSocketClient *)server {
