diff --git a/node_modules/react-native-keyboard-aware-scroll-view/lib/KeyboardAwareHOC.js b/node_modules/react-native-keyboard-aware-scroll-view/lib/KeyboardAwareHOC.js
index 03f46af..3e25918 100644
--- a/node_modules/react-native-keyboard-aware-scroll-view/lib/KeyboardAwareHOC.js
+++ b/node_modules/react-native-keyboard-aware-scroll-view/lib/KeyboardAwareHOC.js
@@ -318,18 +318,29 @@ function KeyboardAwareHOC(
       if (keyboardOpeningTime === undefined) {
         keyboardOpeningTime = this.props.keyboardOpeningTime || 0
       }
-      setTimeout(() => {
-        if (!this.mountedComponent) {
-          return
-        }
-        const responder = this.getScrollResponder()
-        responder &&
-          responder.scrollResponderScrollNativeHandleToKeyboard(
-            reactNode,
-            extraHeight,
-            true
-          )
-      }, keyboardOpeningTime)
+      //setTimeout(() => {
+      //  if (!this.mountedComponent) {
+      //    return
+      //  }
+      //  const responder = this.getScrollResponder()
+      //  responder &&
+      //    responder.scrollResponderScrollNativeHandleToKeyboard(
+      //      reactNode,
+      //      extraHeight,
+      //      true
+      //    )
+      //}, keyboardOpeningTime)
+      if (this.mountedComponent) {
+        UIManager.measureLayout(
+          reactNode,
+          0,
+          () => {},
+          (x, y, width, height) => {
+            findNodeHandle(this._rnkasv_keyboardView), (x, y, width, height) => {
+              this._rnkasv_keyboardView.scrollToPosition(0, parseInt(y) - height - extraHeight, true);
+          }
+        });
+      }
     }
 
     scrollIntoView = async (
