diff --git a/node_modules/react-native-reanimated/lib/module/reanimated2/PlatformChecker.js b/node_modules/react-native-reanimated/lib/module/reanimated2/PlatformChecker.js
index 812f2e8..6d31640 100644
--- a/node_modules/react-native-reanimated/lib/module/reanimated2/PlatformChecker.js
+++ b/node_modules/react-native-reanimated/lib/module/reanimated2/PlatformChecker.js
@@ -38,6 +38,6 @@ export function isWindowAvailable() {
 export function isReducedMotion() {
   return isWeb() ? isWindowAvailable() ?
   // @ts-ignore Fallback if `window` is undefined.
-  !window.matchMedia('(prefers-reduced-motion: no-preference)').matches : false : !!global._REANIMATED_IS_REDUCED_MOTION;
+  !window.matchMedia('(prefers-reduced-motion: no-preference)').matches : false : false;
 }
 //# sourceMappingURL=PlatformChecker.js.map
diff --git a/node_modules/react-native-reanimated/src/reanimated2/PlatformChecker.ts b/node_modules/react-native-reanimated/src/reanimated2/PlatformChecker.ts
index bc22b42..f286c0b 100644
--- a/node_modules/react-native-reanimated/src/reanimated2/PlatformChecker.ts
+++ b/node_modules/react-native-reanimated/src/reanimated2/PlatformChecker.ts
@@ -52,5 +52,5 @@ export function isReducedMotion() {
       ? // @ts-ignore Fallback if `window` is undefined.
         !window.matchMedia('(prefers-reduced-motion: no-preference)').matches
       : false
-    : !!(global as localGlobal)._REANIMATED_IS_REDUCED_MOTION;
+    : false;
 }
