diff --git a/node_modules/react-native-toast-notifications/src/hook/provider.tsx b/node_modules/react-native-toast-notifications/src/hook/provider.tsx
index c342fdb..43a5ce4 100644
--- a/node_modules/react-native-toast-notifications/src/hook/provider.tsx
+++ b/node_modules/react-native-toast-notifications/src/hook/provider.tsx
@@ -1,26 +1,46 @@
+// patched by RewindBitcoin (2 fixes)
+// Patch for {refState && children} (so that it never renders children if for
+// some reason the provider is not ready)
+// More info here: https://github.com/arnnis/react-native-toast-notifications/issues/190#issuecomment-2159901134
+//
+// There is also a patch introducing FullWindowOverlay -  this is done
+// so that this component works in Modals in ios. Took inspìration from:
+// https://github.com/seniv/react-native-notifier/issues/95
+// https://github.com/calintamas/react-native-toast-message/issues/494#issuecomment-2143558602
+
 import React, { FC, useEffect, useRef, useState } from "react";
 import ToastContext, { ToastType } from "./context";
 import Toast, { Props } from "../toast-container";
+import { FullWindowOverlay } from "react-native-screens";
+import { Platform, View } from "react-native";
 
 type PropsWithChildren = Props & {
   children: React.ReactNode;
-}
+};
 
-export let GlobalToast: ToastType
+export let GlobalToast: ToastType;
 
 const ToastProvider: FC<PropsWithChildren> = ({ children, ...props }) => {
   const toastRef = useRef(null);
-  const [refState, setRefState] = useState({});
+  const [refState, setRefState] = useState(null);
 
   useEffect(() => {
     setRefState(toastRef.current as any);
-    GlobalToast = toastRef.current as any
+    GlobalToast = toastRef.current as any;
   }, []);
 
   return (
     <ToastContext.Provider value={refState as any}>
-      {children}
-      <Toast ref={toastRef} {...props} />
+      {refState && children}
+      {Platform.OS === "ios" ? (
+        <FullWindowOverlay>
+          <View pointerEvents={"box-none"}>
+            <Toast ref={toastRef} {...props} />
+          </View>
+        </FullWindowOverlay>
+      ) : (
+        <Toast ref={toastRef} {...props} />
+      )}
     </ToastContext.Provider>
   );
 };
diff --git a/node_modules/react-native-toast-notifications/src/toast.tsx b/node_modules/react-native-toast-notifications/src/toast.tsx
index bbabca6..4c63c00 100644
--- a/node_modules/react-native-toast-notifications/src/toast.tsx
+++ b/node_modules/react-native-toast-notifications/src/toast.tsx
@@ -1,3 +1,6 @@
+// patched by RewindBitcoin
+// 1) shorten the trigger pixels to close the Toast from 50 to 25
+// 2) Add onPanResponderTerminate (read reasoning in the comments)
 import React, { FC, useRef, useEffect, useState } from "react";
 import {
   View,
@@ -225,9 +228,35 @@ const Toast: FC<ToastProps> = (props) => {
         });
       },
       onPanResponderRelease: (_, gestureState) => {
-        if (gestureState.dx > 50) {
+        if (gestureState.dx > 25) {
           panReleaseToRight(gestureState);
-        } else if (gestureState.dx < -50) {
+        } else if (gestureState.dx < -25) {
+          panReleaseToLeft(gestureState);
+        } else {
+          Animated.spring(getPanResponderAnim(), {
+            toValue: { x: 0, y: 0 },
+            useNativeDriver: Platform.OS !== "web",
+          }).start();
+        }
+      },
+      // This fix addresses an issue specific to iOS when using Modals.
+      // If the Toast component is displayed within an iOS Modal and the user
+      // initiates a swipe gesture to dismiss the Toast, there's a potential for
+      // gesture conflicts. When a swipe right/left gesture is started within
+      // the Toast, followed by a swipe up/down gesture in the
+      // Modal, the Modal itself captures the gesture, interrupting the Toast's
+      // gesture handling.
+      // This results in the Toast's PanResponder receiving a termination
+      // request.
+      // The fix below ensures that when such a termination request occurs,
+      // the Toast smoothly returns to its original position or completes,
+      // thus maintaining a consistent and expected user experience.
+      // If not handled, the Toast remains in an indeterminate position since
+      // onPanResponderRelease is never received.
+      onPanResponderTerminate: (_, gestureState) => {
+        if (gestureState.dx > 25) {
+          panReleaseToRight(gestureState);
+        } else if (gestureState.dx < -25) {
           panReleaseToLeft(gestureState);
         } else {
           Animated.spring(getPanResponderAnim(), {
@@ -299,7 +328,7 @@ const Toast: FC<ToastProps> = (props) => {
 
   if (swipeEnabled) {
     animationStyle.transform?.push(
-      getPanResponderAnim().getTranslateTransform()[0]
+      getPanResponderAnim().getTranslateTransform()[0],
     );
   }
 
