diff --git a/node_modules/react-native-modal/dist/.modal.js.swp b/node_modules/react-native-modal/dist/.modal.js.swp
new file mode 100644
index 0000000..5ed1c5d
Binary files /dev/null and b/node_modules/react-native-modal/dist/.modal.js.swp differ
diff --git a/node_modules/react-native-modal/dist/modal.js b/node_modules/react-native-modal/dist/modal.js
index 80f4e75..d5c85d3 100644
--- a/node_modules/react-native-modal/dist/modal.js
+++ b/node_modules/react-native-modal/dist/modal.js
@@ -453,10 +453,17 @@ export class ReactNativeModal extends React.Component {
         if (this.state.isVisible) {
             this.open();
         }
-        BackHandler.addEventListener('hardwareBackPress', this.onBackButtonPress);
+        //BackHandler.addEventListener('hardwareBackPress', this.onBackButtonPress);
+        this.backHandlerSubscription = BackHandler.addEventListener(
+            'hardwareBackPress',
+            this.onBackButtonPress
+        );
     }
     componentWillUnmount() {
-        BackHandler.removeEventListener('hardwareBackPress', this.onBackButtonPress);
+        //BackHandler.removeEventListener('hardwareBackPress', this.onBackButtonPress);
+        if (this.backHandlerSubscription?.remove) {
+            this.backHandlerSubscription.remove();
+        }
         if (this.didUpdateDimensionsEmitter) {
             this.didUpdateDimensionsEmitter.remove();
         }
